test/parallel/test-navigator.js

```diff
@@ -2,8 +2,31 @@
 
 'use strict';
 
-const common = require('../common');
+/* eslint node-core/require-common-first: 0 */
+
 const assert = require('assert');
+
+{
+
+  // Ensures `navigator` has not been evaluated yet
+  assert.strictEqual(require.resolve('../common') in require.cache, false);
+
+  let called = false;
+  const { version, platform, arch } = process;
+  Object.defineProperty(process, 'arch', { get() { called += 'arch|'; return arch; } });
```

Comment:
```suggestion
  Object.defineProperty(process, 'arch', { get: common.mustNotCall() });
```

Reply:
i tried this, but then reverted it, because `common` isn't available yet.